ts/components/SelectLogoutMethod.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Container,
+  Content,
+  H3,
+  Left,
+  List,
+  ListItem,
+  Right,
+  Text,
+  View
+} from "native-base";
+import React from "react";
+import { BackHandler, StyleSheet } from "react-native";
+import { connect } from "react-redux";
+
+import I18n from "../i18n";
+import variables from "../theme/variables";
+
+import IconFont from "../components/ui/IconFont";
+import { LogoutMethod } from "../store/actions/authentication";
+import FooterWithButtons from "./ui/FooterWithButtons";
+
+const styles = StyleSheet.create({
+  content: {
+    padding: variables.contentPadding,
+    paddingTop: 48
+  },
+  separator: {
+    height: 40,
+    width: "100%"
+  },
+  itemLeft: {
+    flexDirection: "column",
+    alignItems: "flex-start"
+  },
+  itemLeftText: {
+    alignSelf: "flex-start"
+  }
+});
+
+type Props = {
+  onCancel: () => void;
+  onMethodSelected: (_: LogoutMethod) => void;
+  header?: React.ReactNode;
+};
+
+/**
+ * A modal that allow the user to select the logout method of choice
+ */
+class SelectLogoutMethod extends React.PureComponent<Props> {
+  private onBackPress = () => {
+    this.props.onCancel();
+    // Returning true is mandatory to avoid the default press action to be
+    // triggered as if the modal was not visible
+    return true;
+  };
+
+  public render() {
+    return (
+      <Container>
```

Comment:
is this a screen? then it should be named `SelectLogoutMethod` and should be under `screens`

Reply:
It is a modal, like _SelectCalendarModal_. Would rather _SelectLogoutMethodModal_ be a more suitable name?

Reply:
yes, thanks